tools/src/argparseUtils.h

```diff
@@ -0,0 +1,54 @@
+#ifndef PODIO_TOOLS_ARGPARSEUTILS_H // NOLINT(llvm-header-guard): folder structure not suitable
+#define PODIO_TOOLS_ARGPARSEUTILS_H // NOLINT(llvm-header-guard): folder structure not suitable
+
+#include <algorithm>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+template <typename... Flags>
+auto findFlags(const std::vector<std::string>& argv, Flags... flags) {
+  return std::find_if(argv.begin(), argv.end(), [&](const auto& elem) { return ((elem == flags) || ...); });
+}
+
+inline std::vector<std::string> splitString(const std::string& str, const char delimiter) {
+  std::vector<std::string> tokens;
+  std::string token;
+  for (char ch : str) {
+    if (ch == delimiter) {
+      if (!token.empty()) {
+        tokens.push_back(token);
+        token.clear();
+      }
+    } else {
+      token += ch;
+    }
+  }
+  if (!token.empty()) {
+    tokens.push_back(token);
+  }
+  return tokens;
+}
```

Comment:
```suggestion
inline std::vector<std::string> splitString(const std::string& str, const char delimiter) {
    std::vector<std::string> result;

    for (const auto part : std::ranges::views::split(str, delimiter)) {
      result.emplace_back(part.begin(), part.end());
    }

    return result;
}
```
There is `std::ranges::views::split` to do exactly what this function does. In principle it should be possible to replace all the occurrences of `splitString` with `std::ranges::views::split` but more code would have to be modified to use views so this is a partial solution.

Reply:
I have made this return a `vector<string_view>` and adapted the `parseSizeOrExit` to take a `string_view`.